### DISCUSSION TITLE
VisionOS Support

### DISCUSSION THREAD

**Discussion by jezell** [zed-industries/zed#25478] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

Would be amazing if Zed supported VisionOS. There are no native IDEs on VisionOS at the moment, so everyone is resorting to virtual desktop. Would be so awesome to have a real IDE there, but VSCode, JetBeans, etc. probably can't ever run there ever since it's built on top of iOS.

**Comment by jackTabsCode** [comment]
I don't see how Zed has any more or less chance of producing a VisionOS app than its competitors. VisionOS ports are accelerated by already having an iPadOS or iOS app-there is little to no overlap, from what I can see, with macOS apps.

**Comment by jezell** [comment]
JVM and Chromium are impossible to run on iOS. Rust is not. Zed has a massive advantage for VisionOS ports, especially with #6222 in the box.

**Comment by unnamedd** [comment]
@jezell Rust is not, yes, you're right on that. However, the GPU library created by Zed folks currently supports only macOS, with some ongoing changes being made to support Linux in the future. Portability to support iOS/iPadOS should be, I believe, a very low priority for them at the moment. Just my two cents.

**Comment by clarkezone** [comment]
I would love iPad support (and by extension AVP)I wonder how much work would be needed to port the GPU library since metal is already xplat.

**Comment by FazalAAli** [comment]
How about LG WebOS support?

**Comment by jezell** [comment]
@unnamedd That's not true. blade-graphics already supports iOS.

**Comment by jezell** [comment]
@clarkezone I believe GPU should already pretty much work since blade-graphics already supports iOS. Input is likely a bigger deal than the rendering. VisionOS has gaze tracking for example, and touch is slightly different than mouse. But even just iPad app working would allow VisionOS support as "Designed for iPad" which would still be awesome.

**Comment by unnamedd** [comment]
> @unnamedd That's not true. blade-graphics already supports iOS.

which unfortunately doesn't mean will run Zed on its full potential. It is very hard to create a project for iOS/iPadOS/visionOS which makes use of things that demanded to be run in background. In case we are talking about having Zed as a simple text editor, then it would work nicely, and I am pretty sure it will be amazing, but even with that, Zed folks will have to make changes in their GPUI to draw the app in the iOS properly (I don't think this would be a big problem for them since Zed works like a game), and also, at the end of all that process, we still have Apple with their Review Process... Well.. Soo many things to be done... I agree with you that it might be awesome, but there is a world of things to be finished before that, but again, this is just my two cents as a developer.

**Comment by dream-dasher** [comment]
Add: iPad is an underexploited niche -- more and more powerful machines, with almost nothing that can run on them.  Ultra portable.  
visionOS can run iPad apps through compatibility mode.

And zed is already being written with a mind toward what to have up-front (on rendering area) and out back (on foreign machine) -- which means using ipad or vision as head to a massive GPU connected server would be more natural there than anywhere else.

It's a strong Niche and one that could be expanded uniquely within. 
(There are literally millions of people who would love to draw math diagrams in notes and code in the same repo, all the while having a local or distant remote machine process a bunch of matrix algebra --- I would love it if Zed put itself on iPad & Vision, and helped set the community up to set it up to succeed.)


For my 2cents: spatial (and I don't just mean apple's term) modes of interacting with data (including code) will be huge boosts and hugely popular once people start to explore them better.
And any code editors that are speedy, know what to render locally vs remotely, and allow flexible front-end interface with raw code string representations will have a huge advantage there.

**Comment by owenpalmer** [comment]
Imagine this setup: Zed on VisionOS with Charachorder's Master Forge keyboard wirelessly connected.
